pgjdbc/src/main/java/org/postgresql/xa/PGXAConnection.java

```diff
@@ -195,12 +197,12 @@ public void start(Xid xid, int flags) throws XAException {
     // It's ok to join an ended transaction. WebLogic does that.
     if (flags == TMJOIN) {
       if (state != State.ENDED) {
-        throw new PGXAException(GT.tr("Transaction interleaving not implemented"),
+        throw new PGXAException(GT.tr("Transaction interleaving not implemented. xid={0}, currentXid={1}", xid, currentXid),
```

Comment:
Just a note: exception message does not reflect the cause. Something like "invalid state" should be used.

Reply:
@vlsi thank you for your fixes. I haven't managed to understand how the call should be compound correctly (eating humble pie).

In the case of the message - do you expect me to change the message? I can only guess what is the purpose. Do you want to change the message to something like `Start called with transaction join flag while transaction is still active. Transaction interleaving not implemented...`, or...?

Reply:
Does the exception message still need changing to "invalid state" ? Or something similar ?

Reply:
"Invalid protocol state requested. Attempted Transaction interleaving is not supported. xid={0}, currentXid={1}"

Reply:
@whitingjr thank you. I changed the PR to the text you suggested. I only wonder if that's right to say that the protocol state is invalid.